PSPs/psp-22.md

```diff
@@ -9,181 +9,674 @@
+A standard interface for WASM contracts Fungible Tokens.
 
-This proposal aims to define the standard token in ink! smart contracts, just like [EIP-20](https://github.com/ethereum/EIPs/edit/master/EIPS/eip-20.md) for Ethereum ecosystem.
+This proposal aims to define the standard fungible token in WASM smart contracts, just like [EIP-20](https://github.com/ethereum/EIPs/edit/master/EIPS/eip-20.md) for Ethereum ecosystem.
 
 ## Importance
-Currently, while there is no standard, every contract will have different signature. Thus, no interoperability is possible. This proposal aims to resolve that by having one **trait**(interface) that shares the
-same **trait naming** between all implementations, as naming of trait affects the identifiers of functions in this trait.
+Currently, while there is no standard, every contract will have different signature. Thus, no interoperability is possible. This proposal aims to resolve that by having one **interface** that shares the
+same **ABI** between all implementations.
 
 ## Implementation
-
-Example implementation:
+Example of Ink! implementation:
 
 - [OpenBrush](https://github.com/Supercolony-net/openbrush-contracts/blob/main/contracts/token/psp20/traits.rs)
 
 ## Motivation
-
-A standard interface allows any Ink! tokens on Polkadot/Kusama to be re-used by other applications: from wallets to decentralized exchanges.
+A standard interface allows any tokens on Polkadot/Kusama to be re-used by other applications: from wallets to decentralized exchanges.
 
 
 ## Motivation for having a standard separate from ERC20
-
-Due to different nature of ink!, the standard should have ink!-specific rules and methods,
+Due to different nature of WASM smart contracts and the difference between EVM and pallet-contract in substrate, the standard should have specific rules and methods,
 therefore PSP-22 differs from ERC-20 in its implementation.
 
-Also, this standard proposal defines an extensive method list in the trait (interface). Unlike ERC20, it includes `increase_allowance` & `decrease_allowance`, and defines metadata fields as part of a separate trait.
-Another difference is that it has `PSP22Receiver` trait, and `on_received` method is called at the end of transfer if the recipient is a contract.
+Also, this standard proposal defines an extensive method list in the(interface. Unlike ERC20, it includes `increase_allowance` & `decrease_allowance`, and defines metadata fields as part of a separate interface.
+Another difference is that it has `PSP22Receiver` interface, and `on_received` method is called at the end of transfer if the recipient is a contract.
+
+# This standard is at ABI level
+
+As contract-pallet in Substrate can execute any WASM contracts, we should not restrain this standard to only Rust & Ink! Framework.
+So that it can be use by any language/framework that compile to WASM.
+The full ABI JSON for interface and events can be found at the top down of this proposal
```

Comment:
```suggestion
As `pallet-contracts` in Substrate can execute any WASM contracts, we should not restrain this standard to only Rust & ink!, so that it can be used by any language/framework that compiles to WASM.
The full ABI JSON for interface and events can be found below.
```